tests/kernel/mem_slab/mslab/src/main.c

```diff
@@ -247,9 +252,10 @@ ZTEST(memory_slab_1cpu, test_mslab)
 
 	TC_PRINT("%s: start to wait for block\n", __func__);
 	k_sem_give(&SEM_REGRESSDONE);    /* Allow helper thread to run part 4 */
-	ret_value = k_mem_slab_alloc(&map_lgblks, &b, K_MSEC(50));
+	ret_value = k_mem_slab_alloc(&map_lgblks, &b, K_MSEC(50 * (TIMEOUT_MULTIPLIER)));
 	zassert_equal(0, ret_value,
-		      "Failed k_mem_slab_alloc, ret_value %d\n", ret_value);
+		      "Failed k_mem_slab_alloc in %d msec., ret_value %d\n",
+		      K_MSEC(50 * (TIMEOUT_MULTIPLIER)), ret_value);
```

Comment:
I... don't understand this?  The test doesn't care about "50ms", it's not looking for a timeout.  It wants the call to **succeed**, and the fact that it times out is an error.  So why make this platform dependent?  Just set it to 100ms for everyone (or K_FOREVER logically, though I'm guessing that the reason this is a timeout is some kind of coverage requirement).

Reply:
yes, this test expects to get a memory block in time which depends on another thread's k_sem_take() and k_mem_slab_free() timings, so to me it looks unsafe to just increase the timeout here because of some particular platform potentially affecting other platforms.
imo, the proposed change with config parameter keeps this test backward compatible and prevents oversight if k_sem_* or k_mem_slab_ delays are really going bad somewhere.

Reply:
How would it be unsafe?  Existing platforms are **passing** this test, so we know they can get the result back in 50ms.   So they can definitely get it back in less than 100ms.

Again, it's just not a test of allocator speed.  Let's not add platform dependence needlessly.  Can we change this to 100ms globally?

Reply:
> How would it be unsafe? Existing platforms are **passing** this test, so we know they can get the result back in 50ms. So they can definitely get it back in less than 100ms.
> 
well, as a 'false positive', e.g. if some existing platform get this test passing with 100ms timeout whereas its usual 40 ms unexpectedly raised to 80ms.

> Again, it's just not a test of allocator speed. Let's not add platform dependence needlessly.
> Can we change this to 100ms globally?

in our case this timeout should be 150ms to pass.